07-routing-gossip.md

```diff
@@ -289,6 +289,12 @@ The receiving node MAY use `rgb_color` and `alias` to reference nodes in interfa
 
 ### Rationale
 
+The `timestamp` field is used by nodes for pruning (either if it's too
+far in the future, or if it's been two weeks with no update), so it
+makes sense to have it be a UNIX timestamp (ie. seconds since UTC
+1970-01-01).  It can't be a hard requirement, however, given the possible case
+of two `channel_update`s within a second.
```

Comment:
I don't think this explains well enough why this can't be made a hard requirement; the reason is because we want nodes to be allowed to broadcast more than one `node_announcement`/`channel_update` a second, into "the far future"? 

Seems like that would be malicious behaviour where we could just ignore the announcements.

Reply:
If you need to send out a temporary disable, you don't have much choice.  It will propagate slowly, of course (each node batches), but it could happen in theory.

Reply:
I guess a node could use frequent `channel_updates` to do some high frequency payments with its immediate channel counter parties. 

But if we allow this (by saying it is not a hard requirement for it to be a unix timestamp), then maybe we shouldn't say that nodes prune based on it being a unix timestamp? 

You might end up with pruning a channel that has issued a lot of updates and now is considered "far into the future", or not pruning a channel that has been inactive for a very long time, but issued a channel update with a future timestamp before it left.

Reply:
Well, we can't prevent some of the foot-gun situations, this is best effort, and we explicitly mention that there is the possibility that frequent updates resulting in far in the future timestamps result in the channel being dropped altogether.

Reply:
Fair enough, thanks!